chore(docs): remove outdated dependency nvim-treesitter

antonk52 opened the pull request:
## Description

nvim-treesitter is not used directly since [10.8](https://github.com/olimorris/codecompanion.nvim/blob/3c9224e3a0084d6c251b9f81d5cb3c6ac12878e6/CHANGELOG.md?plain=1#L1757C197-L1757C292). For this reason there should be no requirement for a user to need the nvim-treesitter plugin as a dependency.

it is only used in minimal repro lua files for installing parsers where I left it unchanged. All other references to the plugin were removed.

I've tested the changes and the chat sidebar as well as `:checkhealth` work the same as current main, except no nvim-treesitter dependency in checkhealth

## Related Issue(s)

issue does not exist

## Screenshots

No visual changes to the plugin

## Checklist

- [x] I've read the [contributing](https://github.com/olimorris/codecompanion.nvim/blob/main/CONTRIBUTING.md) guidelines and have adhered to them in this PR
~~- [ ] I've updated `CodeCompanion.has` in the [init.lua](https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/init.lua#L239) file for my new feature~~ NA
~~- [ ] I've added [test](https://github.com/olimorris/codecompanion.nvim/blob/main/CONTRIBUTING.md#testing) coverage for this fix/feature~~ NA
- [x] I've updated the README and/or relevant docs pages
~~- [ ] I've run `make all` to ensure docs are generated, tests pass and my formatting is applied~~ I tried but there were multiple unrelated changes in the doc/codecompaion.txt so I left it unpushed for now for an easier review

olimorris commented:
Huh?! Are we looking at the same plugin?!

olimorris commented:
Technically, we require a [yaml](https://codecompanion.olimorris.dev/usage/chat-buffer/#settings) and a markdown_inline parser (the latter for images). However, we could always make that explicitly clear in the documentation. I just feel I owe it to any future users to make sure I've handled situations where there isn't a parser, adequately. I'd hate to brick someones chat because I didn't put a simple check in.

Appreciate you digging into this though. I am hugely in favour of removing dependencies across CodeCompanion (not that we have too many now).

antonk52 commented:
Got it, tbh I used it without a yaml parser on and off for the last few months and I didn't notice breakages. Maybe calling it out explicitly in the documentation would be clearer.

> I just feel I owe it to any future users to make sure I've handled situations where there isn't a parser, adequately. I'd hate to brick someones chat because I didn't put a simple check in.

Sure thing, I definitely did not intend to break anything as I've been able to use it without a yaml parser and figured it was unnecessary. 

In this case it also seems like what is actually required is having yaml and markdown_inline parsers and not the nvim-treesitter plugin. As the parsers can also be installed in other ways such as luarocks which is a lot less popular but still an option.

antonk52 commented:
Rebased with no conflicts now, should be good to merge now.

I've also added a line to attach a treesitter to the chat buffer, that required to update some screenshot tests, from the changes you can see that those now properly that the buffer is highlighted using treesitter.
